source/MRViewer/MRRibbonMenu.h

```diff
@@ -120,11 +121,29 @@ class MRVIEWER_CLASS RibbonMenu : public ImGuiMenu
     /// sets flag defining if closing plugin on opening another one is enabled or not
     void setAutoCloseBlockingPlugins( bool value ) { autoCloseBlockingPlugins_ = value; }
 
+    enum class ManualSelectionMode
+    {
+        // Usually triggered by a click without modifiers, selects one object and unselects all others.
+        // The object can be null, then we just unselect all objects.
+        selectOne,
+        // Usually triggered by Ctrl+Click, toggles the selection on one object.
+        toggle,
+    };
+    using ManuallySelectObjectSignal = boost::signals2::signal<bool( Object* object, ManualSelectionMode mode ), StopOnTrueCombiner>;
+    // This is triggered whenever an object is manually selected from the GUI.
+    // Return true to prevent the selection.
+    ManuallySelectObjectSignal manuallySelectObjectSignal;
+
+    // Selects or unselects certain objects in the scene, in a way that can be intercepted (via `manuallySelectObjectSignal`).
+    // This is intended for manual selection by user, not for programmatic selection.
+    // Returns true on success, false if interrupted by the signal.
+    MRVIEWER_API bool manuallySelectObject( Object* object, ManualSelectionMode mode );
```

Comment:
if not supported with scene tree than mb add in comment that it is only applied to pick in name